### ISSUE TITLE
running the example peers.php

### ISSUE THREAD

**Issue by adriavidal** [clue/reactphp-ami#74]
Trying to set up the basic demo before using the library i get these error on peers.php.

php examples/peers.php umyser:mypwas32d1fa@192.168.1.NNN:5038

PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function Clue\React\Ami\Factory::__construct(), 0 passed in /var/www/html/react/examples/peers.php on line 5 and at least 1 expected in /var/www/html/react/vendor/clue/ami-react/src/Factory.php:42
Stack trace:
#0 /var/www/html/react/examples/peers.php(5): Clue\React\Ami\Factory->__construct()
#1 {main}
  thrown in /var/www/html/react/vendor/clue/ami-react/src/Factory.php on line 42

Using version 1.1.0

**Comment by GSM-Call** [comment]
> Trying to set up the basic demo before using the library i get these error on peers.php.
> 
> php examples/peers.php umyser:[mypwas32d1fa@192.168.1.NNN](mailto:mypwas32d1fa@192.168.1.NNN):5038
> 
> PHP Fatal error: Uncaught ArgumentCountError: Too few arguments to function Clue\React\Ami\Factory::__construct(), 0 passed in /var/www/html/react/examples/peers.php on line 5 and at least 1 expected in /var/www/html/react/vendor/clue/ami-react/src/Factory.php:42 Stack trace: #0 /var/www/html/react/examples/peers.php(5): Clue\React\Ami\Factory->__construct() #1 {main} thrown in /var/www/html/react/vendor/clue/ami-react/src/Factory.php on line 42
> 
> Using version 1.1.0



I do have the same problem while trying to test the 3 examples!

**Comment by SimonFrings** [comment]
Hey @adriavidal and @GSM-Call, thanks for bringing this up :+1:

I am not 100% sure, but this looks like a mix up between different project versions. The `Factory` class in v1.1.0 requires a `$loop` parameter, this was changed to an optional `$loop` parameter in preparation for an upcoming v1.2.0 (which is the current main branch and not officially released yet).

I think you're executing the latest version of the `peers.php` example (from the current main branch), which uses the default loop, with the old `Factory` implementation from v1.1.0, which still uses the old required loop. This is why you see a `ArgumentCountError`.

Does this resolves your problem?

**Comment by GSM-Call** [comment]
@SimonFrings OK, noted.

I have replaced the your $factory variable declaration line after the INCLUDE statement in your script and added mine as follows;

```
$loop = React\EventLoop\Factory::create();
$factory = new Clue\React\Ami\Factory($loop);
```

Also,

I have added the following line at the end of the script.

`$loop->run();`

Shall we expect some updates on this project, it is a very useful one for many of us!

**Comment by SimonFrings** [comment]
@clue and I already have the plan to release a 1.2.0 version for this project somewhere in the future, we don't know exactly when this will come as we're currently involved in some other topics. I'll put it on the list 👍

**Comment by SimonFrings** [comment]
I guess the original topic is answered, so I'll close this for now.

**Comment by SimonFrings** [comment]
@adriavidal @GSM-Call To keep you updated, I have just released version 1.2.0 of clue/reactphp-ami. You can start using the new version as of today :+1:
